source/common/router/rds_impl.cc

```diff
@@ -93,6 +93,12 @@ Router::ConfigConstSharedPtr RdsRouteConfigProviderImpl::config() {
 }
 
 void RdsRouteConfigProviderImpl::onConfigUpdate(const ResourceVector& resources) {
+  if (resources.empty()) {
```

Comment:
Doesn't this also apply to LDS/CDS also? I only see RDS/EDS. (Also, again not loving all the duplication but I know we have ticket on trying to consolidate).

Reply:
LDS/CDS empty responses should delete clusters and route configs, since they are always global and provide the source-of-truth on what the valid clusters/route configs are.

Reply:
OK got it.